WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift

```diff
@@ -101,23 +106,51 @@ final class StoreStatsAndTopPerformersViewController: ButtonBarPagerTabStripView
         }
     }
 
-    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        super.collectionView(collectionView, didSelectItemAt: indexPath)
-        let timeRangeTabIndex = indexPath.item
-        guard let periodViewController = viewControllers[timeRangeTabIndex] as? StoreStatsAndTopPerformersPeriodViewController,
-              timeRangeTabIndex != currentIndex else {
+    override func updateIndicator(for viewController: PagerTabStripViewController,
+                                  fromIndex: Int,
+                                  toIndex: Int,
+                                  withProgressPercentage progressPercentage: CGFloat,
+                                  indexWasChanged: Bool) {
+        super.updateIndicator(for: viewController,
+                              fromIndex: fromIndex,
+                              toIndex: toIndex,
+                              withProgressPercentage: progressPercentage,
+                              indexWasChanged: indexWasChanged)
+        // The initially selected tab should be ignored because it should be set after `loadLastTimeRange` in `viewDidLoad`.
+        guard selectedTimeRangeIndex != nil else {
             return
         }
-        saveLastTimeRange(periodViewController.timeRange)
-        syncStats(forced: false, viewControllerToSync: periodViewController)
+        selectedTimeRangeIndex = toIndex
+    }
+
+    func observeSelectedTimeRangeIndex() {
+        let timeRangeCount = timeRanges.count
+        selectedTimeRangeIndexSubscription = $selectedTimeRangeIndex
+            .compactMap { $0 }
+            // It's possible to reach an out-of-bound index by swipe gesture, thus checking the index range here.
+            .filter { $0 >= 0 && $0 < timeRangeCount }
+            .removeDuplicates()
+            // Tapping to change to a farther tab could result in `updateIndicator` callback to be triggered for the middle tabs.
+            // A short debounce workaround is applied here to avoid making API requests for the middle tabs.
+            .debounce(for: .seconds(0.1), scheduler: DispatchQueue.main)
```

Comment:
Just wondering - can 0.1 be too small for slower devices? Should we make it a bit larger (0.2 or 0.3) just in case?

Reply:
makes sense, increased to 0.3s in https://github.com/woocommerce/woocommerce-ios/pull/7650/commits/a6fd2f1442dfed81c923976e65f5ab73a924a454 👍